addons/binding/org.openhab.binding.osramlightify/src/main/java/org/openhab/binding/lightify/internal/LightifyConnector.java

```diff
@@ -0,0 +1,477 @@
+
+    private static final int MAXIMUM_RESENDS = 5;
+
+    private final Logger logger = LoggerFactory.getLogger(LightifyConnector.class);
+
+    private LightifyBridgeHandler bridgeHandler;
+
+    private LightifyTransmitQueue<LightifyMessage> transmitQueue = new LightifyTransmitQueue<>(this);
+
+    private int reconnectAttempts = 1;
+
+    private boolean interrupted = false;
+
+    private Socket socket;
+    private InputStream in;
+    private OutputStream out;
+    private LightifyListPairedDevicesMessage initialPoll;
+
+    private final LightifyListGroupsMessage listGroups = new LightifyListGroupsMessage();
+    private final LightifyListPairedDevicesMessage listPairedDevices = new LightifyListPairedDevicesMessage();
+    private boolean havePoll = false;
+    private long nextPoll;
+    private long pollInterval;
+
+    private int resendCount = 0;
+    private int seqNo = 0;
+
+    private class ExceptionHandler implements Thread.UncaughtExceptionHandler {
+        public void uncaughtException(Thread thread, Throwable throwable) {
+            logger.error("Connector died: ", throwable);
+        }
+    }
+
+    public LightifyConnector(LightifyBridgeHandler bridgeHandler) {
+        setName(getClass().getSimpleName() + "(" + bridgeHandler.getThing().getUID() + ")");
+        this.bridgeHandler = bridgeHandler;
+        setUncaughtExceptionHandler(new ExceptionHandler());
+    }
+
+    @Override
+    public void interrupt() {
+        logger.trace("interrupt received");
+        interrupted = true;
+        super.interrupt();
+    }
+
+    private boolean connectToAddr(String addr) {
+        try {
+            return connectToAddr(InetAddress.getByName(addr));
+        } catch (UnknownHostException e) {
+        }
+
+        return false;
+    }
+
+    private boolean connectToAddr(InetAddress addr) {
+        try {
+            socket = new Socket(addr, LIGHTIFY_GATEWAY_PORT);
+            in = socket.getInputStream();
+            out = socket.getOutputStream();
+            bridgeHandler.getThing().setProperty(PROPERTY_CURRENT_ADDRESS, addr.getHostAddress());
+            return true;
+        } catch (IOException ioe) {
+            disconnect();
+        }
+
+        return false;
+    }
+
+    private synchronized boolean connect() {
+
+        logger.debug("Connecting to Lightify gateway...");
+
+        // Try addresses (if any) found by the mDNS discovery first.
+        InetAddress[] mDNSAddresses = LightifyDiscoveryParticipant.getInetAddressesFor(bridgeHandler.getThing().getUID());
+        if (mDNSAddresses != null) {
+            for (InetAddress addr : mDNSAddresses) {
+                logger.debug("Connecting to Lightify gateway using mDNS address {}:{}", addr.getHostAddress(), LIGHTIFY_GATEWAY_PORT);
+                if (connectToAddr(addr)) {
+                    break;
+                }
+            }
+        }
+
+        // If we still don't have a connection try the statically configured address (if any)
```

Comment:
No. The mDNS results are collected in a map indexed by the same device name as used for the Thing. The definitive address is the one the gateway says it has. Any user-specified address is just a fallback in case you don't have working mDNS between openHAB and the gateway.